pytorch_lightning/strategies/bagua.py

```diff
@@ -0,0 +1,239 @@
+import logging
+import os
+from typing import Any, Dict, List, Optional, Union
+
+import torch
+from torch.nn import Module
+
+import pytorch_lightning as pl
+from pytorch_lightning.overrides.base import _LightningModuleWrapperBase, unwrap_lightning_module
+from pytorch_lightning.plugins.environments.cluster_environment import ClusterEnvironment
+from pytorch_lightning.plugins.io.checkpoint_plugin import CheckpointIO
+from pytorch_lightning.plugins.precision import PrecisionPlugin
+from pytorch_lightning.strategies.ddp import DDPStrategy
+from pytorch_lightning.strategies.strategy import TBroadcast
+from pytorch_lightning.utilities.distributed import ReduceOp
+from pytorch_lightning.utilities.enums import _StrategyType
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.imports import _BAGUA_AVAILABLE
+from pytorch_lightning.utilities.seed import reset_seed
+
+if _BAGUA_AVAILABLE:
+    import bagua.torch_api as bagua
+    from bagua.torch_api.algorithms import Algorithm
+    from bagua.torch_api.algorithms.q_adam import QAdamOptimizer
+    from bagua.torch_api.communication import allreduce_inplace, barrier, broadcast_object, is_initialized
+    from bagua.torch_api.communication import ReduceOp as BaguaReduceOp
+    from bagua.torch_api.data_parallel.distributed import DistributedDataParallel_V1_9_0 as BaguaDistributedDataParallel
+else:
+    BaguaReduceOp = None
+    BaguaDistributedDataParallel = None
+
+log = logging.getLogger(__name__)
+
+
+class LightningBaguaModule(_LightningModuleWrapperBase):
+    def __init__(self, pl_module: "pl.LightningModule") -> None:
+        super().__init__(pl_module)
+        # Bagua use `bagua_module_name` to distinguish different modules
+        self._bagua_module_name = pl_module._get_name() + str(id(pl_module))
```

Comment:
Is there a benefit to using `_get_name`? It's protected so better if we avoid it.

```suggestion
        self._bagua_module_name = f"{pl_module.__class__.__name__}{id(pl_module)}"
```